rfcs/text/i18n.md

```diff
@@ -48,248 +46,137 @@ the bundle size and reducing performance.
+
+export default context.Provider;
 ```
 
 ```js
-// App.js
-import React, { Suspense } from "react";
-import HttpApi from "i18next-http-backend";
-import i18n from "i18next";
-import { initReactI18next, useTranslation } from "react-i18next";
-import Example from "carbon-react/lib/components/Example";
-import I18nProvider from "carbon-react/lib/components/I18nProvider";
-
-i18n
-  .use(initReactI18next)
-  .use(HttpApi)
-  .init({
-    interpolation: {
-      escapeValue: false,
-    },
-    lng: "fr-FR",
-    ns: ["carbon", "app"],
-    backend: {
-      loadPath: "/locales/{{lng}}/{{ns}}.json",
-    }
-  });
+// carbon/components/Pager.js
+import { useI18n } from "../__internal__/I18nContext";
+export default ({
+  pageSize = 25,
+  currentPage = 1,
+  totalRecords = 100,
+  ...props
+}) => {
+  const l = useI18n();
+  const numberOfPages = Math.round(totalRecords / pageSize);
+
+  const pageTranslation = props.pageTranslation || l.Pager.page;
+  const itemsTranslation = props.itemsTranslation || l.Pager.items;
 
-const Page = () => {
-  const { t } = useTranslation('carbon');
-  const { t: appT } = useTranslation('app');
   return (
-    <I18nProvider t={t}>
-      <Example />
-      <Example nextLabel={appT('example.next')} />
-    </I18nProvider>
+    <div className="row">
+      <div className="column">{itemsTranslation(pageSize)}</div>
+      <div className="column">
+        {pageTranslation(currentPage, numberOfPages)}
+      </div>
+    </div>
   );
 };
-
-function App() {
-  return (
-    <Suspense fallback="Loading...">
-      <Page />
-    </Suspense>
-  );
-}
-
-export default App;
-
 ```
 
-Consumers may also choose to bundle the translations.
-
 ```js
-// App.js
-import React, { Suspense } from "react";
-import i18n from "i18next";
-import { initReactI18next, useTranslation } from "react-i18next";
-import Example from "carbon-react/lib/components/Example";
+// consumer/app.js
+import Pager from "carbon-react/lib/components/pager";
 import I18nProvider from "carbon-react/lib/components/I18nProvider";
-import frFr from "carbon-react/lib/translations/fr-FR/carbon.json";
-import AppFrFr from "carbon-react/lib/translations/fr-FR/app.json";
-i18n
-  .use(initReactI18next)
-  .init({
-    resources: {
-      'fr-FR': {
-        carbon: frFr,
-        app: AppFrFr
-      }
-    },
-    lng: "fr-FR",
-    ns: ["carbon", "app"],
-    interpolation: {
-      escapeValue: false
-    }
-  });
-
-const Page = () => {
-  const { t } = useTranslation('carbon');
-    const { t: appT } = useTranslation('app');
-  return (
-    <I18nProvider t={t}>
-      <Example />
-      <Example nextLabel={appT('example.next')} />
-    </I18nProvider>
-  );
-};
+import enXG from "./locales/enXG";
+import { toWords } from "number-to-words";
 
-function App() {
+export default function App() {
   return (
-    <Suspense fallback="Loading...">
-      <Page />
-    </Suspense>
+    <div className="App">
+      {/* Default enGB */}
```

Comment:
is these two styles of comments are on purpose?

Reply:
I cant use `//` because its in the JSX, the other comments are single line so I used `//`.